client/platforms/macos/daemon/wireguardutilsmacos.cpp

```diff
@@ -137,7 +137,13 @@ bool WireguardUtilsMacos::addInterface(const InterfaceConfig& config) {
       if (config.m_killSwitchEnabled) {
         FirewallParams params { };
         params.dnsServers.append(config.m_dnsServer);
-        if (config.m_allowedIPAddressRanges.at(0).toString() == "0.0.0.0/0"){
+
+        QStringList allowedIpsStringList;
+        for (auto allowedIp : config.m_allowedIPAddressRanges) {
```

Comment:
config.m_allowedIPAddressRanges is QList, you can do the same check without this loop,